frontends/smvparser.y

```diff
@@ -487,8 +626,14 @@ basic_expr: constant {
                   smt::Term res = enc.solver_->make_term(smt::Mult, a->getTerm(), b->getTerm());
                   $$ = new SMVnode(res);
               }else{
+                  SMVnode::BVtype bvs_a = a->getBVType();
+                  SMVnode::BVtype bvs_b = b->getBVType();
+                  if(bvs_a != bvs_b){
+                   throw CosaException("Unsigned/Signed unmatch");
+                  } else{
                   smt::Term res = enc.solver_->make_term(smt::BVMul, a->getTerm(), b->getTerm());
-                  $$ = new SMVnode(res);
+                  $$ = new SMVnode(res,bvs_a);
+                  } 
               }
             }
             | basic_expr "/" basic_expr{
```

Comment:
Is there a reason this op isn't called `OP_DIV`?

Reply:
I used symbols instead of OP_ADD, OP_MUL for all arithmetic operations. Do you prefer OP_DIV?

Reply:
Oh no that’s okay — as long as there’s a pattern. Sounds good!